agenta-backend/agenta_backend/tasks/evaluations.py

```diff
@@ -32,65 +32,81 @@
 
 @shared_task(queue="agenta_backend.tasks.evaluations.evaluate")
 def evaluate(
-    app_data: dict,
-    new_evaluation_data: dict,
-    evaluation_id: str,
+    app_id: str,
+    variant_id: str,
+    evaluators_config_ids: List[str],
     testset_id: str,
+    evaluation_id: str,
+    rate_limit_config: Dict[str, int],
 ):
+    """
+    Evaluate function that performs the evaluation of an app variant using the provided evaluators and testset.
+    Saves the results in the Database
+
+    Args:
+        app_id (str): The ID of the app.
+        variant_id (str): The ID of the app variant.
+        evaluators_config_ids (List[str]): The IDs of the evaluators configurations to be used.
+        testset_id (str): The ID of the testset.
+        rate_limit_config (Dict[str,int]): See LLMRunRateLimit
+
+    Returns:
+        None
+    """
+
     loop = asyncio.get_event_loop()
 
     try:
+        # Fetch data from the database
         loop.run_until_complete(DBEngine().init_db())
-        app = loop.run_until_complete(fetch_app_by_id(app_data["_id"]))
-        evaluation = NewEvaluation(**new_evaluation_data)
-        variant_id = str(evaluation.variant_ids[0])
+        app = loop.run_until_complete(fetch_app_by_id(app_id))
         app_variant_db = loop.run_until_complete(fetch_app_variant_by_id(variant_id))
         app_variant_parameters = app_variant_db.config.parameters
-
-        if (
-            not app_variant_parameters
-            or "inputs" not in app_variant_parameters
-            or not app_variant_parameters["inputs"]
-        ):
-            loop.run_until_complete(
-                update_evaluation(evaluation_id, {"status": "EVALUATION_FAILED"})
-            )
-            return
```

Comment:
I think the if is important here and the data we check are important for the llm app, consider having it back

Reply:
The check is totally wrong. inputs does not have to be in the app_variant_parameters. There is not even a need to have app_variant_parameters for an app to work.